draft-mouris-cfrg-mastic.md

```diff
@@ -661,15 +680,154 @@ malicious Aggregator.
+the attempt, such as the time that elapsed between when the connection attempt
+began and the error was observed (e.g., {{W3C23}}, Section 7). Clients may also
+report successful connection attempts to give the server a sense of the error
+rate. The exact client behavior is determined by the reporting policy specified
+by the server (see {{W3C23}}, Section 5.1).
+
+NEL data is privacy-sensitive for two reasons. First, it exposes information
+that the server would not otherwise have access to, which can be abused to probe
+the client's network configuration as described in [{{W3C23}}, Section 9].
+Second, for operational reasons, the reporting endpoint may be organizationally
+separated from the server (i.e., run on different cloud infrastructures),
+leading to an increased risk of the client's browsing history being exposed
+(e.g., in a data breach).
+
+**Private NEL with Mastic.** MPC helps mitigate these risks by revealing to the
+endpoint only the information it needs to fulfill its service level objectives.
+This means, of course, we must be satisfied with limited functionality.
+Fortunately, Mastic allows us to preserve the most important functionality of
+NEL while minimizing privacy loss.
+
+We consider here a simplified version of NEL where each client reports a tuple
+`(dom, err)` consisting of a domain name dom (e.g., `example.com`) and a value
+err that represents an error (e.g., `dns.unreachable`) or an indication that no
+error occurred (e.g., `ok`). Notably, this can be easily extended in Mastic to
+represent more elaborate metrics. e.g., where each weight includes the time it
+took each browser to report the error (and the aggregate is the average error
+reporting time), user agent (browser type and version), etc. However, our main
+goal is to understand 1) the distribution of errors and 2) which domains are
+impacted.
+
+We expect there to be a large number of distinct domain names (millions in the
+case of content delivery networks) and only a small number of error variants
+(the NEL spec {{W3C23}} defines 30 variants). The following Mastic parameters
+are suitable for this application.
+
+* **Inputs:** Each input `alpha` encodes the domain `dom` truncated to `n = 256`
+bits, which is sufficient to represent most of the domains on the internet
+{{MST24}}, {{BBCGGI21}}. Domains that are shorter than `n` bits are padded with
+0s.
+* **Weights:** Each weight `beta` represents the error variant `dom`. To compute
+the distribution of errors, we encode each error variant as a distinct bucket of
+a histogram so that `[1, 0, 0, ...]` represents `ok`, `[0, 1, 0, ...]`
+represents `dns.unreachable`, `[0, 0, 1, ...]` represents
+`dns.name_not_resolved`, and so on. There are 30 such variants (see {{W3C23}},
+Section 6), so the language `L` of valid weights is exactly the set of length-30
+vectors over `F` containing all 0s except for a single 1.
+* **Ordering:** Our `order(·)` function computes the ratio of reports with `err
```

Comment:
Note that unicode charcters, like "·", should not appear in internet drafts. With the exception of authors, it's a good idea to make sure every character is ASCII.
```suggestion
vectors over `F` containing all 0s except for a single 1.

* Our `order` function computes the ratio of reports with `err
```